design/src/rfcs/rfc0017_impl_async_read_for_bytestream.md

```diff
@@ -0,0 +1,198 @@
+RFC: Implementing `tokio::io::AsyncRead` for `aws_smithy_http::byte_stream::ByteStream`
+=================================================
+
+> Status: RFC
+
+Customers have expressed the desire that [`aws_smithy_http::byte_stream::ByteStream`][ByteStream] should implement the [`tokio::io::AsyncRead`][AsyncRead] trait. This would enable them to make use of existing tools for decompressing and buffering data:
+
+```rust
+let data: Box<dyn AsyncRead> = Box::new(some_bytestream);
+// Read a gzipped text file line by line
+for line in BufReader::new(GzipDecoder::new(data))).lines() {
+  println!("{line}");
+}
+```
+
+Users currently have the option of wrapping `ByteStream` in a newtype and implementing the `AsyncRead` trait themselves. This is not a good customer experience for reasons of ergonomics and performance that we'll cover later in this RFC.
+
+## Implementing `AsyncRead` for a newtype
```

Comment:
Why can't we implement it directly on `ByteStream`? If we did, a buffered version of `ByteStream` can be created using [`tokio::io::BufReader`](https://docs.rs/tokio/latest/tokio/io/struct.BufReader.html).